src/compiler/parser.ts

```diff
@@ -5650,9 +5650,27 @@ namespace ts {
             return finishNode(node);
         }
 
+        function tryParseStringConstructorDeclaration(node: ConstructorDeclaration): ConstructorDeclaration | undefined {
+            return tryParse(() => {
+                const stringLiteral = parseLiteralNode();
+                if (isStringLiteral(stringLiteral)
+                    && stringLiteral.text === "constructor"

```

Comment:
Is the string literal allowed to contain escapes?

Reply:
Yes, escapes are handled in the lexical grammar. `text` is the unescaped version, so it should be fine.

Reply:
I asked because I wasn't sure if the spec allowed escapes in the string literal. Tested it with Chrome and it recognizes it as constructor.

Reply:
Sorry, I’m not completely following—can you show me exactly what you tried with Chrome?

Reply:
To test whether ES6 classes allow escapes in the sting literal `"constructor"` I executed the following code:
```js
new (class C {"\x63onstructor"(param) {console.log(param)}})(1)
```

I was just curious because `"use strict"` is only recognized if it does not contain escapes.

Reply:
I actually don’t see anything in the grammar that prohibits that from being a proper constructor, so maybe we should be looking at escaped text. But I could be missing something. Was looking around [this area](https://www.ecma-international.org/ecma-262/9.0/index.html#sec-static-semantics-constructormethod) and [here](https://www.ecma-international.org/ecma-262/9.0/index.html#prod-LiteralPropertyName). I’m leaning toward assuming Node and Chrome are doing the correct thing.

Reply:
Handled escaped constructors in the latest commit. I think this is ready now.